File: ejercicios java semana 2/ejercicio 5.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// -------------------------
// FUNCIÓN PRINCIPAL
// -------------------------
function latencyAuditor(listaTiempos, umbral) {

    // 1️⃣ Calcular promedio
    const suma = listaTiempos.reduce((acc, val) => acc + val, 0);
    const promedio = suma / listaTiempos.length;

    // 2️⃣ Detectar pico extremo
    let picoExtremo = false;

    for (let tiempo of listaTiempos) {
        if (tiempo >= 3 * promedio) {
            picoExtremo = true;
            break;
        }
    }

    // 3️⃣ Condición final
    const alerta = promedio > umbral || picoExtremo;

    return alerta;
}

// -------------------------
// PROGRAMA PRINCIPAL
// -------------------------

rl.question("Ingresa los tiempos separados por comas: ", function(entrada) {

    const listaTiempos = entrada
        .split(",")
        .map(x => parseFloat(x.trim()))
        .filter(x => !isNaN(x));

    rl.question("Ingresa el umbral crítico: ", function(inputUmbral) {

        const umbral = parseFloat(inputUmbral);

        if (isNaN(umbral)) {
            console.log("Umbral inválido");
            rl.close();
            return;
        }

        const resultado = latencyAuditor(listaTiempos, umbral);

        console.log("¿Se activa alerta?:", resultado);

        rl.close();
    });
});

File: ejercicios python 2/ejercicio 5.py
def latency_auditor(lista_tiempos, umbral):
    
    # 1️ Calcular promedio
    promedio = sum(lista_tiempos) / len(lista_tiempos)
    
    # 2️ Bandera para detectar picos extremos
    pico_extremo = False
    
    for tiempo in lista_tiempos:
        if tiempo >= 3 * promedio:
            pico_extremo = True
            break
    
    # 3 Condición final de alerta
    alerta = promedio > umbral or pico_extremo
    
    return alerta


# -------------------------
# PROGRAMA PRINCIPAL
# -------------------------

# Pedir lista al usuario
entrada = input("Ingresa los tiempos de respuesta separados por comas: ")
lista_tiempos = [float(x) for x in entrada.split(",")]

umbral = float(input("Ingresa el umbral crítico: "))

resultado = latency_auditor(lista_tiempos, umbral)

print("¿Se activa alerta?:", resultado)
